### PR TITLE
removed deprecated translation/translator methods

### PR DESCRIPTION
**Pull request by OskarStark**
I am targeting this branch, because I removed already deprecated methods.

### CODE REVIEW

**Comment by OskarStark** [comment]
Or do we keep them to avoid merge conflicts @greg0ire ?

**Comment by greg0ire** [comment]
It might be best to merge this at the latest possible date, but if we do that, then we have a big cleanup to do… IMO we can merge and fight the conflicts. If we make sure master is updated regularly, it shouldn't be an issue.

**Comment by OskarStark** [comment]
Travis should be green now

File: Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -572,22 +560,7 @@ public function getDataSourceIterator()
...
-            // No translation key exists
-            if ($transLabel == $label) {
-                $fields[$key] = $field;
-            } else {
-                $fields[$transLabel] = $field;
-            }
-        }
-
-        return $this->getModelManager()->getDataSourceIterator($datagrid, $fields);
+        return $this->getModelManager()->getDataSourceIterator($datagrid, $this->getExportFields());
```
Comment (OskarStark) [inline Admin/AbstractAdmin.php]: @greg0ire not sure if this change is fine ❓

Comment (greg0ire) [inline Admin/AbstractAdmin.php]: I think @core23 wrote the instructions. Might be fine, but please double-check together

**Comment by bgaleotti** [comment]
I think lines [1](https://github.com/OskarStark/SonataAdminBundle/blob/remove-deprecated-code/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L211) and [2](https://github.com/OskarStark/SonataAdminBundle/blob/remove-deprecated-code/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L261) should also be removed, and maybe [this](https://github.com/sonata-project/SonataAdminBundle/blob/master/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L279-L281) too?

**Comment by OskarStark** [comment]
> I think lines 1 and 2 should also be removed, and maybe this too?

thank you for your feedback @bgaleotti, I think you are right, i removed these lines of code.

**Comment by OskarStark** [comment]
I am not sure if we need to remove the whole translator stuff from the AdminExtractor too, or if we need to adjust it to use the default translator if available via injection, but this would be a BC break, which was not deprecated before

**Comment by jordisala1991** [comment]
If we want to remove that, it is better to deprecate it first on 3.x. Because of the merge conflicts.

I would say, lets keep this PR for the things that are already deprecated, to get it merged faster

**Comment by OskarStark** [comment]
if i remove the last commit with the extractor, the tests are failing due to: `$admin->getTranslator()` in the extractor 😢

**Comment by jordisala1991** [comment]
Oh... then we just need to deprecate it first on 3.x ? It is suposed to be deprecated right?

**Comment by OskarStark** [comment]
This PR is OnHold until #4424 is merged!

**Comment by OskarStark** [comment]
I currently have no time to work on it

**Pull request closed without merging**